_posts/2022-06-08-packaging-functions-with-helm.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "Packaging OpenFaaS functions with Helm"
+description: "Learn how to package and deploy your OpenFaaS functions with Helm."
+date: 2022-06-08
+image: /images/2022-06-packaging-functions-with-helm/background.jpg
+categories:
+- helm
+- functions
+- kubernetes
+- enterprise
+author_staff_member: han
+---
+
+Learn how to package and deploy your OpenFaaS functions with Helm.
+
+## The OpenFaaS Operator
+As far as we know, the most common way to deploy functions right now is by using the `faas-cli` commands. The CLI uses the OpenFaaS REST API to deploy functions. In modern production environments you might want to use a GitOps approach to deploy build artefacts from your Continuous Integration pipeline to your cluster. At this time, two of the most popular open-source projects for GitOps are [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) and [Flux](https://fluxcd.io/). Both of these projects are designed to apply Kubernetes YAML files and can not directly interact with the OpenFaaS REST API. As an alternative to deploying functions through the REST API, OpenFaaS supports a second op mode using a Kubernetes [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
+
+![Using the CRD to deploy functions](/images/2021-06-kubectl-functions/operator-crd.png)
+> Using the CRD to deploy functions
+
+## Deploying functions using the Function CRD
```

Comment:
This would be a good place to show an example.

Reply:
Do you want me to include an example of what a Function Custom Resource looks look?

Or how generate and deploy it because that is covered in the next chapter.

Reply:
Yes include the example that I linked you to that says "extended example"